src/scenes/Activities/ActivityHistory/EventRows/index.js

```diff
@@ -63,7 +63,7 @@ class EventRow extends Component {
             <TableCell className={classes.summaryRowCell}>{getShortLocalDateTimeString(createdTime)}</TableCell>
             <TableCell>{getTxTypeString(type, locale, localeMessages)}</TableCell>
             <NameLinkCell className={classes.viewEventLink} clickable={topic && topic.address} onClick={this.handleClick(txid, topic && topic.address)}>
-              {name || (topic && topic.name)}
+              {name === 'Invalid' && localizedInvalid !== undefined ? localizedInvalid[intl.locale.slice(0, 2)] : name || (topic && topic.name)}
```

Comment:
move this logic into model (Oracle) and pass in the locale.

good practice is to keep the UI as dumb as possible. get data and show. in this case, let the model do the heavy lifting.

Reply:
I actually can use the original code, since you issue this as a bug in sprint board, "activity history name should be event name, not option name"

Reply:
Ah yes you are right. This is showing the wrong thing here. Transaction History in the event detail page is where you should show the localized Invalid (if necessary)